pro_1.py

```diff
@@ -0,0 +1,92 @@
+import json
+from re import A
+import re
+
+database_file=open("database.txt","r+")
+dict_n=json.load(database_file)
+def loged_in():
+    print("succes")
+
+def login():
+    print("login page")
+    return
+def signup():
+    pattern1="\w{1,}.\w{1,}@\w{1,}.com"
+    pattern2="^.*(?=.{8,})(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[@#$%^&+=]).*$"
+    print("signup page")
+    name=input("enter ur full name ")
+    i=0
+    while(len(name)==0):
+        name=input("re enter ur name")
+        
+        i=i+1
+        if i==2:
+            exit()
+    birth_date=input("enter ur birth date")
+    i=0
+    while(len(birth_date)==0):
+        birth_date=input("re enter ur birth_date")
+        
+        i=i+1
+        if i==2:
+            exit()
+    email=input("enter ur email")
+    i=0
+    while(len(email)==0):
+        email=input("re enter ur email")
+        
+        i=i+1
+        if i==2:
+            exit()
+    for i in dict_n:
+     if i==email:
```

Comment:
this is cool but checking if the key is used before is done like this, it is always better to use the predefined methods as they are optimized most of the times.

if dict_n.get(i) is not None: